mobile/mobile-core/devices-service.ts

```diff
@@ -378,6 +401,13 @@ export class DevicesService implements Mobile.IDevicesService {
 		}
 
 		return this.executeOnAllConnectedDevices(action, canExecute);
+}
+
+	private isApplicationInstalledOnDevice(deviceIdentifier: string, appIdentifier: string): IFuture<boolean> {
+		return ((): boolean => {
```

Comment:
why? In fact I prefer to have it as the method is used for Proton only. The idea is that this future will be waited in the decorator and if it fails, the Promise will be rejected.
In case I remove it and `this.getDeviceByIdentifier` fails, the public method will fail before returning future for this device and it will hang :)